x/mongo/driver/topology/pool_test.go

```diff
@@ -296,10 +296,6 @@ func TestPool(t *testing.T) {
 
 			c, err = p.get(context.Background())
 			noerr(t, err)
-			gen = atomic.LoadUint64(&c.generation)
```

Comment:
Previously, the pool incremented its generation number on `disconnect` to indicate that all checked out connections should be considered stale as they return themselves to the pool, so this assertion made sense. Because the new map type tracks multiple generation numbers, it instead uses an atomic boolean value to store connected/disconnected state to make decisions in its `stale` method. To reflect that, this assertion has been replaced by checks of the generation map's state.